service/controller/chart/resource/release/resource.go

```diff
@@ -183,52 +184,56 @@ func isEmpty(c ReleaseState) bool {
 	return equals(c, ReleaseState{})
 }
 
+// isReleaseFailed checks if the release is failed. If the values or version
+// has changed we return false and will attempt to update the release. As this
+// may fix the problem.
 func isReleaseFailed(current, desired ReleaseState) bool {
-	if isEmpty(current) {
-		return false
-	}
+	result := false
 
-	// Values have changed so we will try to update the release.
-	if current.ValuesMD5Checksum != desired.ValuesMD5Checksum {
-		return false
-	}
+	if !isEmpty(current) {
+		// Values have changed so we should try to update even if the release
+		// is failed.
+		if current.ValuesMD5Checksum != desired.ValuesMD5Checksum {
+			return false
+		}
```

Comment:
@yasn77 @tomahawk28 This just needed a tweak.

The early return is needed so the log messages are correct. We should log `the release does not have to be updated` rather than `the release is in status DEPLOYED and cannot be updated.`

But it's still a lot cleaner which make me happy. Thx!